components/Chips/src/MDCChipView.m

```diff
@@ -394,14 +412,48 @@ - (UIColor *)inkColorForState:(UIControlState)state {
 - (void)setInkColor:(UIColor *)inkColor forState:(UIControlState)state {
   _inkColors[@(state)] = inkColor;
 
+  NSNumber *rippleState = [self rippleStateForControlState:state];
+  if (rippleState) {
+    [self.rippleView setRippleColor:inkColor forState:rippleState.integerValue];
+  }
+
   [self updateInkColor];
+  [self updateRippleColor];
 }
 
 - (void)updateInkColor {
   UIColor *inkColor = [self inkColorForState:self.state];
   self.inkView.inkColor = inkColor ? inkColor : self.inkView.defaultInkColor;
 }
 
+- (void)updateRippleColor {
+  UIColor *rippleColor = [self inkColorForState:self.state];
+  // MDCStatefulRippleView sets the ripple color internally when its state changes.
+  // If that specific state isn't supported by the stateful ripple, then we directly set the
+  // ripple view's color to the requested color.
+  if (![self rippleStateForControlState:self.state]) {
+    self.rippleView.rippleColor =
+        rippleColor ?: [UIColor colorWithWhite:1 alpha:MDCChipViewRippleDefaultOpacity];
+  }
+}
+
+- (NSNumber *)rippleStateForControlState:(UIControlState)state {
+  // We check to see if MDCRippleState conforms to a UIControlState and return it, otherwise
+  // we return nil for non-supported ripple states.
+  switch (state) {
+    case UIControlStateNormal:
+      return [NSNumber numberWithInteger:MDCRippleStateNormal];
+    case UIControlStateHighlighted:
+      return [NSNumber numberWithInteger:MDCRippleStateHighlighted];
+    case UIControlStateSelected:
+      return [NSNumber numberWithInteger:MDCRippleStateSelected];
```

Comment:
Does the [Material Activated state](https://material.io/design/interaction/states.html#activated) need to be handled in this logic? Since a Chip can be either Selected (Filter Chips) or Activated (Choice Chips), when should `UIControlStateSelected` be mapped to the Activated Material (Ripple) state?

Reply:
As far as I'm aware, we don't have the "activated" state as part of the ChipView or ChipCollectionCell API, but rather we treat it identically to Selected. Once we do support it, we may probably need to add support for it in the implementation file and maybe also support it in Ripple's states.

Reply:
Please provide guidance as part of your states work regarding future work on the component to provide new states.

Reply:
Sounds good. We can add these in a followup PR.